src/otx/algorithms/classification/adapters/mmcls/models/classifiers/custom_image_classifier.py

```diff
@@ -300,6 +304,57 @@ def extract_feat(self, img):
         ReciproCAMHook,
     )
 
+    def _extract_vit_feat(model, x):
+        """Modified forward from mmcls.models.backbones.vision_transformer.VisionTransformer.forward()."""
+        B = x.shape[0]
+        x, patch_resolution = model.backbone.patch_embed(x)
+
+        # stole cls_tokens impl from Phil Wang, thanks
+        cls_tokens = model.backbone.cls_token.expand(B, -1, -1)
+        x = torch.cat((cls_tokens, x), dim=1)
+        x = x + resize_pos_embed(
+            model.backbone.pos_embed,
+            model.backbone.patch_resolution,
+            patch_resolution,
+            mode=model.backbone.interpolate_mode,
+            num_extra_tokens=model.backbone.num_extra_tokens,
+        )
+        x = model.backbone.drop_after_pos(x)
+
+        if not model.backbone.with_cls_token:
+            # Remove class token for transformer encoder input
+            x = x[:, 1:]
+
+        feat = None
+        layernorm_feat = None
+        for i, layer in enumerate(model.backbone.layers):
+            if i == len(model.backbone.layers) - 1:
+                layernorm_feat = layer.norm1(x)
+
+            x = layer(x)
+
+            if i == len(model.backbone.layers) - 1 and model.backbone.final_norm:
+                x = model.backbone.norm1(x)
+
+            if i in model.backbone.out_indices:
+                B, _, C = x.shape
+                if model.backbone.with_cls_token:
+                    patch_token = x[:, 1:].reshape(B, *patch_resolution, C)
+                    patch_token = patch_token.permute(0, 3, 1, 2)
+                    cls_token = x[:, 0]
+                else:
+                    patch_token = x.reshape(B, *patch_resolution, C)
+                    patch_token = patch_token.permute(0, 3, 1, 2)
+                    cls_token = None
+                if model.backbone.output_cls_token:
+                    feat = [patch_token, cls_token]
+                else:
+                    feat = patch_token
+        backbone_feat = feat
+        if model.with_neck:
+            feat = model.neck(feat)
+        return feat, backbone_feat, layernorm_feat
```

Comment:
Could you rename feat->logits if this tensor represents logits?

Reply:
Here, feat is not a logits yet. Logits are obtained at the next step `logit = self.head.simple_test(feat)`